### ISSUE TITLE
Changing Attribute Set on a Product remembers previous Attribute Set values.

### ISSUE THREAD

**Issue by macrodesignuk** [magento/magento2#25476]
### Preconditions (*)

1. Version 2.3.2, 2.3-develop
2.

### Steps to reproduce (*)

1. Create new product attribute Type dropdown called shoe_size, create some values 10,11,12 etc
2. Create a new Attribute Set Shoes and assign the attribute shoe_size.
3. Create a new Attribute Set Shirts, but don't assign any new attributes other than the default.
4. Create a new product and assign the attribute set for shoes, selecting a shoe size 10.
5. Save the product
6. Go back and edit the new product, changing the attribute set to Shirts.
7. Create a product price rule where conditions are for shoe_size 10, and 10% discount.
8. You may need to apply rule and re-index. 
9. View the product on the front end of the site and you'll see the product has a 10% discount.

### Expected result (*)

1. When changing a products Attribute Set, the previous inapplicable attributes should be removed from the database.
2. Alternatively, the attributes for Price Rule conditions should not be applied to a product where the attribute is not part of the selected attribute set.

### Actual result (*)

1. A product is discounted if it previously had an attribute set assigned, that had an attribute that is currently discounted.
2. The discounted attribute does not need to exist in the currently selected attribute set for that product, but the discount is still applied as the original entries were not removed from the database.

I've created an SQL query to list the product attributes assigned to the product and if you enter the new product entity ID in, you'll see the shoe size attribute is still assigned to it.
```
SELECT cpei.entity_id,ea.attribute_code 
FROM `catalog_product_entity_int` cpei
INNER JOIN eav_attribute ea ON ea.attribute_id = cpei.attribute_id
WHERE cpei.entity_id= [product entity id here]
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22917` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by deep701** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @deep701. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by deep701** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @deep701. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by sdzhepa** [comment]
We are not able to reproduce this issue on the latest `2.4-develop` branch.
Probably it was already fixed in some other PR

Need re-check and close it or confirm as Reproduced on 2.4.x

**Comment by magento-engcom-team** [comment]
@m2-backlog[bot] Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@m2-backlog[bot] Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by engcom-Delta** [comment]
Issue is not reproducible on the latest 2.4-develop
**Manual testing scenario:**
1. Under Stores / Attributes / Products / Create New Attribute Type drop down called shoe_size
2. Add some values to it Eg 10, 11, 12
![image](https://user-images.githubusercontent.com/51681379/99267271-6601d400-282c-11eb-9af2-43814f699d11.png)
3. Under Storefront Properties tab, set Use for Promo Rule conditions to Yes
4. Under Stores / Attributes / Attribute set  Add Attribute set called  Shoes 
5. Assign shoe_size attribute to this Attribute set 
![image](https://user-images.githubusercontent.com/51681379/99267542-bda03f80-282c-11eb-93d2-0a6cf13a73d9.png)

6. Create a new attribute set called Shirts and leave it default 
7. Create a new Product and add select Shoes as Attribute set and 10 as shoe_size 
![image](https://user-images.githubusercontent.com/51681379/99267423-99446300-282c-11eb-952d-e357f9b244da.png)
8. Edit the new product once again and change the Attribute set to Shirts
9. Create a New Rule under Marketing / Catalog Price Rule / 
10. Set Condition to shoe_size is 10 and Action to 10% discount 
![image](https://user-images.githubusercontent.com/51681379/99267671-ec1e1a80-282c-11eb-9405-454c743aa2d7.png)
11. Reindex and clean cache if needed 
12. Navigate to frontend and check the price of the product 

**Result:**
:heavy_check_mark: Price without discount is displayed for the product 
![image](https://user-images.githubusercontent.com/51681379/99267223-571b2180-282c-11eb-8788-beaa58e82030.png)

**Comment by magento-engcom-team** [comment]
Hi @macrodesignuk. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-25157` was closed as `non-reproducible` in `2.4-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.
